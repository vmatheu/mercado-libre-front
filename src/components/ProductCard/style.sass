.mcl-product-card 
	margin: 0 0 10px 10px 
