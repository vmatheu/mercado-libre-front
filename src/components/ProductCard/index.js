import React from 'react';
import { string, number, bool } from 'prop-types';
import './style.sass';

const ProductCard =
  ({ id, title, currency, amount, decimals, picture, condition, freeShipping }) => (
    <div className="ard mb-4 shadow-sm mcl-product-card">
      <div className="card-header">
        <h4 className="my-0 font-weight-normal">{title}</h4>
      </div>
      <div className="card-body">
        <h3 className="card-title pricing-card-title">${amount}</h3>
        <img alt={`imagen del producto ${title}`} src={`http://${picture}`} />
        <h6>{freeShipping}</h6>
      </div>
    </div>
);

ProductCard.propTypes = {
  id: string.isRequired,
  title: string.isRequired,
  currency: string.isRequired,
  amount: number.isRequired,
  decimals: number.isRequired,
  picture: string.isRequired,
  condition: string.isRequired,
  freeShipping: bool.isRequired,
};

export default ProductCard;
