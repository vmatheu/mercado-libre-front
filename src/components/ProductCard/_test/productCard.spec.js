import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ProductCard from '..';

describe('<ProductCard/>', () => {
  describe('should be product card', () => {
    it('should have the same ui structure with one product', () => {
      const product = {
        id: "id_test",
        title: "Relojes Por Mayor Digitales Imantados Unisex ( 5 Unidades )",
        amount: 3087.5,
        currency: "CLP",
        decimals: 20,
        picture: "http://miImagenNBonita.cl",
        condition: "condition",
        freeShipping: true
      }

      const wrapper = shallow(<ProductCard {...product} />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
