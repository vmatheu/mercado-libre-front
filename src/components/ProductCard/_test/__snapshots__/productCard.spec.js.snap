// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard/> should be product card should have the same ui structure with one product 1`] = `
<div
  className="ard mb-4 shadow-sm mcl-product-card"
>
  <div
    className="card-header"
  >
    <h4
      className="my-0 font-weight-normal"
    >
      Relojes Por Mayor Digitales Imantados Unisex ( 5 Unidades )
    </h4>
  </div>
  <div
    className="card-body"
  >
    <h3
      className="card-title pricing-card-title"
    >
      $
      3087.5
    </h3>
    <img
      alt="imagen del producto Relojes Por Mayor Digitales Imantados Unisex ( 5 Unidades )"
      src="http://http://miImagenNBonita.cl"
    />
    <h6>
      <Component />
    </h6>
  </div>
</div>
`;
